### DISCUSSION TITLE
How to resize 2 tiled windows together?

### DISCUSSION THREAD

**Discussion by dcorking** [rxhanson/Rectangle#363] [Q&A] [answered]
I tiled 2 windows, left half and right half, then I dragged their shared border.

However only the active window resizes. For example if the right window is active, and I drag the border to the left, it overlaps the left window.

Is there a way for my drag to shrink the left window at the same time, so the two windows remain tiled? If so how? (I'm happy to be recommended to use another app alongside Rectangle to achieve this.)

**Comment by dcorking** [comment, marked as answer]
After a little more research it seems possible that the answer will be to make my 2 monitors into separate Spaces, and log out :
> make sure “Displays have separate Spaces” is selected.

https://support.apple.com/en-gb/HT204948

**Reply by rxhanson** [reply]
Right, so split view built into macOS is my recommendation for this functionality. You can assign keyboard shortcuts to these actions by following steps in this post: https://medium.com/ryan-hanson/going-without-a-3rd-party-window-manager-in-macos-catalina-4bd270b29245

While it is possible to implement this behavior as a 3rd party, the result is not desirable in my opinion. The api available to 3rd parties does not allow programmatic window resizing from the left edge to be smooth, and you end up with very distracting jerky window edge movements. Last I tried Amethyst, it exhibited this behavior.

**Reply by dcorking** [reply]
Thank you for writing that article: the built in vertical split tile behaviour is quite nice (a bit like Gnome 3 and Windows 10 IIUC) though hard to discover. I tried it and confirm that Quartz only allows a vertical split: no quarters and I miss my multi-monitor Spaces. I'll bind some key chords to Quartz's tiling, but I'll keep Rectangle running and use it when I need more complex layouts.

I'll also try yabai at some point and report back.
